arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -33,104 +190,356 @@ class CommonStressorParams:
         if self.cpu_load is not None:
-            result = result + "cpu-load {}\n".format(self.cpu_load)
+            result += f"cpu-load {self.cpu_load}\n"
+        if self.cpu_method is not None:
+            result += f"cpu-method {CpuMethod(self.cpu_method)}\n"
         return result
 
 
 @dataclass
 class VmStressorParams(CommonStressorParams):
     vm_bytes: typing.Annotated[
-        str,
-        schema.name("VM Memory"),
-        schema.description("Amount of memory a single VM stressor will use"),
-    ]
-
-    mmap: typing.Annotated[
         typing.Optional[str],
-        schema.name("Mmap"),
-        schema.description("Number of stressors per CPU"),
+        schema.id("vm-bytes"),
+        schema.name("VM Memory Bytes"),
+        schema.description("mmap N bytes per vm worker, the default is 256MB"),
+    ] = None
+
+    vm_ops: typing.Annotated[
+        typing.Optional[int],
+        schema.id("vm-ops"),
+        schema.name("VM Operations"),
+        schema.description("Stop vm workers after N bogo operations"),
+    ] = None
+
+    vm_hang: typing.Annotated[
+        typing.Optional[int],
+        schema.id("vm-hang"),
+        schema.name("VM Hang"),
+        schema.description(
+            "Sleep N seconds before unmapping memory, the default is zero seconds"
+        ),
+    ] = None
+
+    vm_keep: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("vm-keep"),
+        schema.name("VM Keep"),
+        schema.description(
+            "Do not continually unmap and map memory, just keep on re-writing to it"
+        ),
+    ] = None
+
+    vm_locked: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("vm-locked"),
+        schema.name("VM Locked"),
+        schema.description(
+            "Lock the pages of the mapped region into memory using mmap "
+            "MAP_LOCKED (since Linux 2.5.37)"
+        ),
+    ] = None
+
+    vm_method: typing.Annotated[
+        typing.Optional[VmMethod],
+        schema.id("vm-method"),
+        schema.name("VM Method"),
+        schema.description(
+            "Specify a vm stress method; by default, all the stress methods "
+            "are exercised sequentially"
+        ),
+    ] = VmMethod.ALL
+
+    vm_populate: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("vm-populate"),
+        schema.name("VM Populate"),
+        schema.description(
+            "populate (prefault) page tables for the memory mappings; "
+            "this can stress swapping"
+        ),
+    ] = None
+
+    def to_jobfile(self) -> str:
+        result = f"vm {self.workers}\n"
+        if self.vm_bytes is not None:
+            result += f"vm-bytes {self.vm_bytes}\n"
+        if self.vm_ops is not None:
+            result += f"vm-ops {self.vm_ops}\n"
+        if self.vm_hang is not None:
+            result += f"vm-hang {self.vm_hang}\n"
+        if self.vm_keep is True:
+            result += "vm-keep\n"
```

Comment:
This is obviously legal Python, and it forms a structure which is nicely parallel to that of the previous lines, but comparing what should be a boolean value against `True` is not "Pythonic".  This should just be
```python
        if self.vm_keep:
            result += "vm-keep\n"
```
since we're not implementing tri-state logic here; that is, if `vm_keep` should happen to be `None` instead of `False`, the code will still function correctly.

(Arguably, we should also be using an implicit boolean-style test of the other options, such that a value of `None` would be the not-`True` case, but, since some of these values might legitimately be zero, which would also evaluate to not-`True`, the explicit test for `None` is better, but that's because these values are not booleans.)